### PR TITLE
SALTO-4769: Add properties & push status to ProfileMapping instances (Okta)

### PR DESCRIPTION
**Pull request by shir-reifenberg**
Add push status info for ProfileMapping instances

---

changes:
1. Add deployment for ProfileMapping instances (only modifications)
2. Fetch properties for profile mapping which also contains info re push status of each property

---

_Release Notes_: 

_Okta_adapter_:
- Fetch properties push status in ProfileMapping instances
- Support deployments of ProfileMapping

---

_User Notifications_: 

_Okta_adapter_:
- properties push status will be added to ProfileMapping instances

### CODE REVIEW

File: packages/okta-adapter/src/config.ts [hunk 1]
```diff
@@ -1224,6 +1225,16 @@ const DEFAULT_TYPE_CUSTOMIZATIONS: OktaSwaggerApiConfig['types'] = {
       fieldsToOmit: DEFAULT_FIELDS_TO_OMIT.concat({ fieldName: '_links' }),
       fieldsToHide: [{ fieldName: 'id' }],
     },
+    deployRequests: {
+      // only modificaiton is supported
```
Comment (netama) [inline packages/okta-adapter/src/config.ts:1229]: 1. typo nit
```suggestion
      // only modification is supported
```
2. we can still support additions if we first find the correct mapping id, no? (in a deploy that creates a new app won't we also want to support customizing the mapping?)

File: packages/okta-adapter/src/filters/profile_mapping_properties.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import { FETCH_CONFIG } from '../config'
+
+const log = logger(module)
+
+const getProfileMapping = async (
+  mappingId: string,
+  client: OktaClient
+): Promise<Values> => (await client.getSinglePage({
+  url: `/api/v1/mappings/${mappingId}`,
+})).data as Values[]
```
Comment (netama) [inline packages/okta-adapter/src/filters/profile_mapping_properties.ts:30]: hmm is it `Values` or `Values[]`? (and should we verify something about the structure?)

**Review by netama — COMMENTED** [review]
looks good :) added some questions

File: packages/okta-adapter/src/config.ts (hunk 1 above)
Comment (shir-reifenberg) [reply packages/okta-adapter/src/config.ts:1229]: yes thats a good point, I'll add it

File: packages/okta-adapter/src/config.ts [hunk 3]
```diff
@@ -1219,11 +1220,27 @@ const DEFAULT_TYPE_CUSTOMIZATIONS: OktaSwaggerApiConfig['types'] = {
   },
   ProfileMapping: {
     transformation: {
-      idFields: ['source.name', 'target.name'],
+      idFields: ['&source.id', '&target.id'],
```
Comment (shir-reifenberg) [inline packages/okta-adapter/src/config.ts:1223]: Not super related to this PR, but we need to change id fields. name might repeat per app (for example if there are two salesforce apps connected), might do this in a separate PR as we still have some missing references there that might cause internal ids in elemIds

File: packages/okta-adapter/test/filters/add_alias.test.ts [hunk 4]
```diff
@@ -25,8 +24,6 @@ describe('addAliasFilter', () => {
   const groupType = new ObjectType({ elemID: new ElemID(OKTA, GROUP_TYPE_NAME) })
   const appType = new ObjectType({ elemID: new ElemID(OKTA, APPLICATION_TYPE_NAME) })
   const profileMappingType = new ObjectType({ elemID: new ElemID(OKTA, PROFILE_MAPPING_TYPE_NAME) })
-  const config = { ...DEFAULT_CONFIG }
-  config[FETCH_CONFIG].includeGroupMemberships = true
```
Comment (shir-reifenberg) [inline packages/okta-adapter/test/filters/add_alias.test.ts:29]: not related to this PR but I noticed its redundant

File: packages/okta-adapter/src/filters/profile_mapping_addition.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        && isAdditionChange(change)
+        && getChangeData(change).elemID.typeName === PROFILE_MAPPING_TYPE_NAME
+    )
+
+    const deployResult = await deployChanges(
+      relevantChanges.filter(isInstanceChange),
+      async change => deployProfileMappingAddition(change, client, config[API_DEFINITIONS_CONFIG])
+    )
+
+    return {
```
Comment (netama) [inline packages/okta-adapter/src/filters/profile_mapping_addition.ts:81]: from IRL - let's add a TODO for the deletions

**Review by netama — APPROVED** [review]
